benchmarks/inference/llm-bench/src/llm_bench/clients/azure_ml_client.py

```diff
@@ -0,0 +1,69 @@
+import json
+import requests
+from typing import Any, Dict
+
+from .base import BaseClient
+from ..config import BaseConfigModel
+from ..prompt import Prompt
+from ..response import Response
+from ..status import Status
+
+
+class AzureMLClientConfig(BaseConfigModel):
+    # TODO: Add descriptions for each field
+    api_url: str = ""
+    api_key: str = ""
+    deployment_name: str = ""
+
+
+class AzureMLClient(BaseClient):
+    def __init__(self, config: AzureMLClientConfig) -> None:
+        super().__init__(config)
+        self.api_url = config.api_url
+        self.api_key = config.api_key
+        self.deployment_name = config.deployment_name
+
+    # TODO: Implement these methods
+    def start_service(self) -> Status:
+        pass
+
+    def stop_service(self) -> Status:
+        pass
+
+    def prepare_request(self, prompt: Prompt) -> Dict[str, Any]:
+        if prompt.streaming:
+            raise ValueError("AzureMLClient does not support streaming prompts.")
+
+        headers = {
+            "Content-Type": "application/json",
+            "Authorization": ("Bearer " + self.api_key),
+            "azureml-model-deployment": self.deployment_name,
+        }
+        pload = {
+            "input_data": {
+                "input_string": [
+                    prompt.text,
+                ],
+                "parameters": {
+                    "max_tokens": prompt.max_new_tokens,
+                    "return_full_text": prompt.return_full_text,
+                },
+            }
+        }
+        return {"url": self.api_url, "headers": headers, "json": pload, "timeout": 180}
+
+    def send_request(self, request_kwargs: Dict[str, Any]) -> Any:
+        while True:
+            try:  # Sometimes the AML endpoint will return an error, so we send the request again
+                response = requests.post(**request_kwargs)
+                output = json.loads(response.content)
```

Comment:
Do we need the code handling different response formatting as we did in the original benchmarking code?
https://github.com/microsoft/DeepSpeedExamples/blob/cce62236a2c8f52d5548f310e64ee09ed2785416/benchmarks/inference/mii/src/client.py#L160C1-L169C22

We added these `try-excepts` due to deployments, particularly HF ones, returning the response in a different format depending on the model.